processor/src/tests/literal/mod.rs

```diff
@@ -19,6 +51,141 @@ mod bitcoin {
     check::<IsTrue<{ Bitcoin::DUST >= bitcoin_serai::wallet::DUST }>>();
   }
 
+  #[test]
+  fn test_receive_data_from_input() {
+    let docker = spawn_bitcoin();
+    docker.run(|ops| async move {
+      let btc = bitcoin(&ops).await;
+
+      // generate a musig address to receive the funds
+      let mut keys = frost::tests::key_gen::<_, <Bitcoin as Network>::Curve>(&mut OsRng)
+        .remove(&Participant::new(1).unwrap())
+        .unwrap();
+      <Bitcoin as Network>::tweak_keys(&mut keys);
+      let group_key = keys.group_key();
+      let address = <Bitcoin as Network>::external_address(group_key);
+
+      // btc key pair to spend from.
+      let private_key = PrivateKey::new(SecretKey::new(&mut rand_core::OsRng), BNetwork::Regtest);
+      let public_key = PublicKey::from_private_key(SECP256K1, &private_key);
+      let main_addr = BAddress::p2pkh(&public_key, BNetwork::Regtest);
+
+      // make some funds to spent
+      let new_block = btc.get_latest_block_number().await.unwrap() + 1;
+      btc
+        .rpc
+        .rpc_call::<Vec<String>>("generatetoaddress", serde_json::json!([100, main_addr]))
+        .await
+        .unwrap();
+
+      // create a scanner
+      let db = MemDb::new();
+      let mut scanner = new_scanner(&btc, &db, group_key, &Arc::new(Mutex::new(true))).await;
+
+      // make a transfer instruction & hash it for script.
+      let serai_address = insecure_pair_from_name("dadadadada").public();
+      let message = Shorthand::transfer(None, serai_address.into()).encode();
+      let mut data = Sha256::engine();
+      data.input(&message);
+
+      // make the output script => `OP_SHA256 PUSH MSG_HASH OP_EQUAL`
+      let script = ScriptBuf::builder()
+        .push_opcode(OP_SHA256)
+        .push_slice(Sha256::from_engine(data).as_byte_array())
+        .push_opcode(OP_EQUAL)
+        .into_script();
```

Comment:
The output script also needs to be *concatenated* with a traditional spend script. Ideally, the `scriptPubKey` is written `msg_script || spend_script` and the sig `msg || script_sig` (where, of course, `script_sig` is a `vec` of witness items, so `msg` would simply be the first).

Else there's a DoS.